### DISCUSSION TITLE
feature request: Alignment of modifier objects.

### DISCUSSION THREAD

**Discussion by LukaBronzovic** [OrcaSlicer/OrcaSlicer#1671] [Ideas]
Hi,

So far I really like both Bambu and Orca slicers. I use Orca. 

When we add modifiers or load them from a file, we can only move them around using mouse and that is not precise enough and in some cases can lead to undesired results.

One thing that I would really like to see is the alignment controls of the modifiers. 

Examples: 
1. Align a selected modifier plane to a selected part plane.
2. Align a selected modifier axis to the selected part axis. 

In both cases the part would probably stay as is and modifier object would move/rotate etc. 
Modifier's alignments can be stored as degrees of movement/rotation freedom limitations.  
Different 3D modeling tools call it Constraints, Joints, etc.

I guess one could design all the modifiers in 3D modeling tools and load them later in the slicer. In that case I guess not more than 3 axis alignments would be required.
Perhaps, some direction swapping and offsets would be nice to give users a full control.

Kind regards,

Luka

**Comment by coltography** [comment]
Really surprised this isn't already a feature considering you can align models. 

For example I just tried to align an SVG onto the center of a circle and it's nearly impossible to get it perfect. Would love to at least be able to align to center of the build plate along with the model.

**Comment by hcoohb** [comment]
Very interested in this too.
Or even being able to set the modifiers based on the lowest z point coordinate would be much easier than to set the coordinate of the middle of the part!

**Comment by vl-pub** [comment]
it'll be usefull
